File: model/src/test/java/com/lightbend/model/SimpleFactoryResolver.java
package com.lightbend.model;

import java.util.HashMap;
import java.util.Map;

// Factories resolver for Wine data
public class SimpleFactoryResolver implements ModelFacroriesResolver {

    private static final Map<Integer, ModelFactory> factories = new HashMap<Integer, ModelFactory>() {
        {
            put(Modeldescriptor.ModelDescriptor.ModelType.TENSORFLOW.getNumber(), SimpleTensorflowModelFactory.getInstance());
            put(Modeldescriptor.ModelDescriptor.ModelType.TENSORFLOWSAVED.getNumber(), SimpleTensorflowBundleModelFactory.getInstance());
            put(Modeldescriptor.ModelDescriptor.ModelType.PMML.getNumber(), SimplePMMLModelFactory.getInstance());
        }
    };

    @Override
    public ModelFactory getFactory(int type) {
        return factories.get(type);
    }
}

File: model/src/test/scala/com/lighbend/modelServer/SimpleFactoryResolver.scala
package com.lighbend.modelServer

import com.lightbend.model.modeldescriptor.ModelDescriptor
import com.lightbend.modelServer.model.{ModelFactory, ModelFactoryResolver}

class SimpleFactoryResolver extends ModelFactoryResolver{

  private val factories = Map(
    ModelDescriptor.ModelType.PMML.value -> SimplePMMLModel,
    ModelDescriptor.ModelType.TENSORFLOW.value -> SimpleTensorflowModel,
    ModelDescriptor.ModelType.TENSORFLOWSAVED.value -> SimpleTensorflowBundleModel
  )

  override def getFactory(`type`: Int): Option[ModelFactory] = factories.get(`type`)
}